CodingChallenges/CC_046_Asteroids/Processing/CC_046_Asteroids/CC_046_Asteroids.pde

```diff
@@ -0,0 +1,79 @@
+// Daniel Shiffman
+// http://codingtra.in
+// http://patreon.com/codingtrain
+// Code for: https://youtu.be/hacZU523FyM
+// Processing transcription: Chuck England
+
+import java.util.*;
```

Comment:
Is this required?

Reply:
Short anser: Yes

Long drawn out answer: This is required to support the generic ArrayList/List implementations. Potentially I could not use the generics, but I would have to look to see if the non-generic is available or needed and import. In addition, the added casting required for the non-generics tends to confuse folks more than the generics do. For some of the things, I might only need an array. But for any of the places where I am calling add(), I need to be able to dynamically allocate space. So, that either becomes, create a new array of the new size and copy all of the existing elements to it plus the new elements, or simply call add on one of the dynamically sizable list types.